scripts/upload-artifacts/upload_artifacts.py

```diff
@@ -20,31 +17,76 @@ def run_command(command):
     return result.stdout
 
 
-# function to parse image URL and extract image name and tag
 def parse_image_url(image_url, destination_registry):
-    from docker_image import reference
-    # Initialize image_name and image_tag
-    ref = reference.Reference.parse(image_url)
-    image_tag = ref['tag'] if ref['tag'] else 'latest'
-    
-    image_name = ""
-    if ref['name'] is None:
-        raise ValueError(f"Invalid image URL: {image_url}")
-    else:
-        image_split = ref['name'].split('/')
-        if len(image_split) == 1:
-            image_name = image_split[0]
+    """
+    /*
+    * Normalize the image name to a standard format
+    * docker.io -> library
+    * replace registry_url with destination_registry
+    * parse_image_url("nginx","tfy.jfrog.io/tfy-images") tfy.jfrog.io/tfy-images/nginx
+    * parse_image_url("nginx:1.1") tfy.jfrog.io/tfy-images/nginx:1.1
+    * parse_image_url("nginx:latest") tfy.jfrog.io/tfy-images/nginx:latest
+    * parse_image_url("nginx:1.1@sha256:asdasdasdsadsad")tfy.jfrog.io/tfy-images/nginx:1.1
+    * parse_image_url("docker.io/truefoundrycloud/nginx:1.1@sha256:asdasdasdsadsad") tfy.jfrog.io/tfy-images/truefoundrycloud/nginx:1.1
+    * parse_image_url("docker.io/nginx") tfy.jfrog.io/tfy-images/library/nginx:latest
+    */
+    """
+    # Split the image URL by ':' to separate the tag if present
+    image_parts = image_url.split(":")
+
+    # If a tag exists, use it, otherwise set 'latest' as default
+    image_tag = image_parts[1].split('@')[0] if len(image_parts) > 1 else "latest"  # Remove SHA digest
```

Comment:
what happens here if `tag` is not present? Wouldn't the split function throw an error?

Reply:
I tried an image with no tag, It picked the latest, it doesn't throw error but return a list of 1 element

Reply:
yes so `image_parts` will have length 1 and `image_parts[1]` will be `None`?

Reply:
I'd suggest to stick with the registry parsing that the library provides or these corner cases will make things complicated in the longer run

Reply:
yes, image_parts[1]  if tag is missing It will always fallback to `latest`, would never be > 1 due to the scheme
if

Reply:
made the changes